src/AzSK.ADO/Framework/Core/STMapping/AzSKADOServiceMapping.ps1

```diff
@@ -421,10 +490,54 @@ class AzSKADOServiceMapping: CommandBase
+                                    $inputbody = "{'contributionIds':['ms.vss-serviceEndpoints-web.service-endpoints-details-data-provider'],'dataProviderContext':{'properties':{'serviceEndpointId':'$($servConnID)','projectId':'$($this.projectId)','sourcePage':{'url':'$($sourcePageUrl)','routeId':'ms.vss-admin-web.project-admin-hub-route','routeValues':{'project':'$($this.ProjectName)','adminPivot':'adminservices','controller':'ContributedPage','action':'Execute'}}}}}" | ConvertFrom-Json
+                                    $responseObj = [WebRequestHelper]::InvokePostWebRequest($apiURL, $inputbody); 
+
+                                    if ([Helpers]::CheckMember($responseObj, "dataProviders") -and $responseObj.dataProviders."ms.vss-serviceEndpoints-web.service-endpoints-details-data-provider") 
+                                    {
+                                        $serviceConnEndPointDetail = $responseObj.dataProviders."ms.vss-serviceEndpoints-web.service-endpoints-details-data-provider"
+                                        if ($serviceConnEndPointDetail.serviceEndpoint.type -eq "azurerm")
+                                        {
+                                            $subscriptionID = $serviceConnEndPointDetail.serviceEndpoint.data.subscriptionId;
+                                            Write-Progress -Activity 'Fetching Service Id from Azure Data explorer...' -CurrentOperation $serviceConnEndPointDetail.serviceEndpoint.name;
+
+                                            # call data studio to fetch azure subscription id and servce id mapping
+                                            $apiURL = "https://datastudiostreaming.kusto.windows.net/v2/rest/query"                                                                                                                
+                                            $inputbody = '{"db": "Shared","csl": "DataStudio_ServiceTree_AzureSubscription_Snapshot | where SubscriptionId contains ''{0}''", "properties": {"Options": {"query_language": "csl","servertimeout": "00:04:00","queryconsistency": "strongconsistency","request_readonly": false,"request_readonly_hardline": false}}}'
+                                            $inputbody = $inputbody.Replace("{0}", $subscriptionID)  
+
+                                            # generate access token with datastudio api audience                      
+                                            $accessToken =[ContextHelper]::GetGraphAccessToken($false,$true)
+                                            $header = @{
+                                                "Authorization" = "Bearer " + $accessToken
+                                            }
+                                            $responseObj = [WebRequestHelper]::InvokeWebRequest([Microsoft.PowerShell.Commands.WebRequestMethod]::Post,$apiURL,$header,$inputbody,"application/json; charset=UTF-8");                               
```

Comment:
all this api call logic can be keep in a method and call that method here as it is repeating code